MITADS/opensubtitles_exporter.py

```diff
@@ -0,0 +1,354 @@
+  [ u'4°' , u'quarto' ],
+  [ u'5°' , u'quinto' ],
+  [ u'49°' , u'quarantanovesimo' ],
+  
+  ##numero, metto uno spazio perchè certe volte il simbolo è attaccato al numero
+  [ u'n°' , u'numero ' ],  
+
+  ##importi in valuta, sono presenti   
+  [ re.compile('\$\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\1 dollari' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*\$'), r'\1 dollari' ],
+  [ re.compile('(₤|£)\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\2 lire' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*₤'), r'\1 lire' ],
+  [ re.compile('(€)\s*([0-9]+[.,]{0,1}[0-9]*)'), r'\2 euro' ], 
+  [ re.compile('([0-9]+[.,]{0,1}[0-9]*)\s*€'), r'\1 euro' ],
+  ##trim spazi
+  [ re.compile('\s+'), u' ' ]
+]
+
+
+def maybe_normalize(value, mapping=mapping_normalization):
+  for norm in mapping:
+    if type(norm[0]) == str:
+      value = value.replace(norm[0], norm[1])
+    elif isinstance(norm[0], Pattern):
+      try:
+        value = norm[0].sub(norm[1], value)
+      except Exception as e:
+        raise(e)
+    else:
+      print('UNEXPECTED', type(norm[0]), norm[0])
+
+  ##remove markup tags...(non sono presenti nel dataset )
+  #value = loads(value)
+
+  return value
+
+def line_not_relevant(text):
+  ##implementare logiche di esclusione righe
+  if(len(text)<2):
+    return True
+
+  return False
+
+
+def clear_text(text):
+  ##text = re.sub(r'[^\x00-\x7F]+',' ', text)  mi toglie anche le È
+
+  #PER I CARATTERI UNICODE ho usato libreria unidecode, questi non servono più
+  #text = text.replace('\u200b',' ')## ZERO WIDTH SPACE
+  #text = text.replace(u'ε','e') ##GREEK SMALL LETTER EPSILON
+  #text = text.replace(u'♪','') ## char ♪
+  #text = text.replace(u'♫','') ## char ♫
+  #text = text.replace(u'¶','') ## char ♫
+  
+  #text = text.sub(r'[^\x00-\x7F]+',' ', text_ret)
+  
+  text = text.strip()
+
+  return text
+
+def normalize_text(text):
+
+  ##SE E SOLO SE  tutto il testo è tutto upper case allora lo porto a lower case
+  if(text.isupper()):
+    text = text.lower()
+
+  ##se una sentence contiene una sola parola con caratteri speciali 
+  # ALLORA tutta la sentence viene esclusa (fa  parte dei titoli di testa)
+  
+  if(not ' ' in text and 
+    (   '_' in text
+     or '@' in text
+     or '}' in text     
+     or '+' in text
+     or '{' in text)):
+     return ''
+  ##caratteri che mi invalidano riga a prescindere dal numero di word
+  if('®' in text
+     or '©' in text
+     or '±' in text):
+    return ''
+
+  #normalization (replace and sub regex)
+  text = maybe_normalize(text,mapping_normalization)
+
+  ######da spostare sulla clear?? vedere se spostando la chiamata non abbiamo side effect 
+  # unidecode normalizza i caratteri speciali ascii (altrimenti andrebbe in errore la fase di scrittura su file con utf-8)
+  text = unidecode(text)
+  #################
+
+  text = text.strip()
+
+  if(len(text)==0):
+    return ''
+  #################################
+  ##other normalization step
+
+  ##rimuovo il punto a pice all'inizio della sentence
```

Comment:
Maybe we can do a regex for that?

Reply:
I think it is more efficient to do it this way